### PR TITLE
Integration testing for assignment management

### PR DESCRIPTION
**Pull request by tanmoysrt**
#### Describe the changes you have made in this PR -
- when user is a primary mentor
   - delete assignment
   - is able to edit assignment when assignment is not closed 
   -  is not able to edit assignment when assignment is closed
   - is able to close assignment
   - is able to reopen assignment
- when user is mentor
    - delete assignment
    - creates assignment
    - does not create assignment when name is blank
    - is able to edit assignment when assignment is not closed
    - is not able to edit assignment when assignment is close
    - is not able to close assignment
    - is able to reopen assignment

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

File: spec/system/assignment_spec.rb [hunk 1]
```diff
@@ -41,8 +53,92 @@
...
+      visit group_path(@group)
+      click_link "Reopen"
+      visit group_path(@group)
+      expect(page).to have_text("Close")
+    end
+  end
+
+  context "when user is mentor" do
+    it "creates assignment" do
+      sign_in @mentor
```
Comment (smritigarg) [inline spec/system/assignment_spec.rb:102]: **Suggestion** Probably put this in a `before` block?

File: spec/system/assignment_spec.rb [hunk 2]
```diff
@@ -57,6 +153,31 @@
       click_link "View"
       check_show_page(name, deadline, description, "Input, Button")
     end
+
+    it "is not able to edit assignment when assignment is closed" do
+      sign_in @mentor
+      @assignment = FactoryBot.create(:assignment, group: @group, status: "closed")
```
Comment (smritigarg) [inline spec/system/assignment_spec.rb:159]: **Suggestions** Avoid using instance variables in specs, and use let wherever possible https://github.com/rubocop/rspec-style-guide#instance-variables

File: spec/system/assignment_spec.rb [hunk 3]
```diff
@@ -5,9 +5,11 @@
 describe "Assignments", type: :system do
   before do
     @primary_mentor = FactoryBot.create(:user)
+    @mentor = FactoryBot.create(:user)
```
Comment (smritigarg) [inline spec/system/assignment_spec.rb:8]: **Suggestion** Shall we move the instance variable definitions to let block instead ?https://github.com/rubocop/rspec-style-guide#instance-variables

**Review by smritigarg — COMMENTED** [review]
Thanks for the change @tanmoysrt Just a few suggestions for your consideration. 🏓

**Comment by tanmoysrt** [comment]
@smritigarg sure ! will update the PR based on the suggestion

**Comment by tanmoysrt** [comment]
@smritigarg updated the PR accordingly the suggestion. Please check out

**Review by smritigarg — APPROVED** [review]
Changes look good to me. Approving the same, thanks @tanmoysrt . @tachyons May be we can merge these changes whenever you get some time.

File: spec/system/assignment_spec.rb [hunk 4]
```diff
@@ -3,18 +3,112 @@
...
 describe "Assignments", type: :system do
+  let(:primary_mentor) { FactoryBot.create(:user) }
+  let(:group) { FactoryBot.create(:group, primary_mentor: primary_mentor) }
+  let(:mentor) do
+    user = FactoryBot.create(:user)
+    FactoryBot.create(:group_member, group: group, user: user, mentor: true)
+    user
+  end
+  let(:member) do
+    user = FactoryBot.create(:user)
```
Comment (tachyons) [inline spec/system/assignment_spec.rb:14]: We can do something like this to simplify 
```suggestion
    user = FactoryBot.create(:user, group_members:  [FactoryBot.create(:group_member, group: group)])
```

**Pull request merged**